Index: statistics.php
===================================================================
--- statistics.php	(revision 17)
+++ statistics.php	(working copy)
@@ -18,203 +18,63 @@
  *Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
-require("db.php");
-require("boccelib.php");
+require("start.php");
 require("statslib.php");
-require("header.php");
-require("side.php");
-?>
-<div class="body">
-<h1>Office Bocce Scores</h1>
-<br />
-<table width="100%" class="stats">
-<tr>
-<th>Name</th>
-<th>Record (W-L)</th>
-<th>Win Percentage</th>
-<th>Deuces Per Game</th>
-<th>Deuces Against Per Game</th>
-<th>Points per Game</th>
-<th>PA per Game</th>
-<th>Current Streak</th>
-</tr>
 
-<?
+$page->assign('subtitle',"Office Bocce Scores");
 
+$players = Array();
+
 db_open();
 
 foreach (all_players() as $player) {
-  print "<tr>";
-
-  print "<th><a href=\"personal.php?id=$player\">";
-  print player_name($player)."</a></th>\n";
-
-  print "<td>".player_total_wins($player)."&nbsp;-&nbsp;";
-  print player_total_losses($player)."</td>\n";
-
-  print "<td>".format_percent(player_overall_win_percentage($player)).
-    "</td>\n";
-
-  print "<td>".format_average(player_overall_deuces_per_game($player)).
-    "</td>\n";
-
-  print "<td>".format_average(player_overall_deuces_against_per_game($player)).
-    "</td>\n";
-
-  print "<td>".format_average(player_overall_points_per_game($player)).
-    "</td>\n";
-
-  print "<td>".format_average(
-    player_overall_points_against_per_game($player))."</td>\n";
-
-  print "<td>".player_overall_current_streak($player)."</td>\n";
-
-  print "</tr>\n";
+	$players[] = Array(
+		'id' => $player,
+		'name' => player_name($player),
+		'wins' => player_total_wins($player),
+		'loses' => player_total_losses($player),
+		'overall_win_perc' => format_percent(player_overall_win_percentage($player)),
+		'overall_deuces_per_game' => format_average(player_overall_deuces_per_game($player)),
+		'overall_deuces_against_per_game' => format_average(player_overall_deuces_against_per_game($player)),
+		'overall_points_per_game' => format_average(player_overall_points_per_game($player)),
+		'overall_points_against_per_game' => format_average(player_overall_points_against_per_game($player)),
+		'overall_current_streak' => player_overall_current_streak($player),
+		// second table
+		'overall_points_per_game_as_loser' => format_average(player_overall_points_per_game_as_loser($player)),
+		'overall_points_against_per_game_as_winner' => format_average(player_overall_points_against_per_game_as_winner($player)),
+		'overall_bruise_perc' => format_percent(player_overall_bruise_percentage($player)),
+		'overall_bruise_attempts_per_game' => format_average(player_overall_bruise_attempts_per_game($player)),
+		'overall_rounds_per_game' => format_average(player_overall_rounds_per_game($player)),
+		'overall_deuces_per_round' => format_percent(player_overall_deuces_per_round($player)),
+		'overall_double_deuce_to_turkey_conv_perc' => format_percent(player_overall_double_deuce_to_turkey_conversion_percentage($player)),
+		'overall_coinflip_win_perc' => format_percent(player_overall_coinflip_win_percentage($player)),
+		// third table
+		'total_deuces' => player_total_deuces($player),
+		'total_points_scored' => player_total_points_scored($player),
+		'total_points_scored_against' => player_total_points_scored_against($player),
+		'total_turkeys' => player_total_turkeys($player),
+		'total_tetrises' => player_total_tetrises($player),
+		'total_bruises_succ' => player_total_bruises_successful($player),
+		'total_bruises_unsecc' => player_total_bruises_unsuccessful($player),
+		'total_shutouts' => player_total_shutouts($player),
+		'total_been_shutouts' => player_total_been_shutouts($player));
 }
 
-?>
-</table>
+$page->assign('players', $players);
 
-<br />
-
-<table width="100%" class="stats">
-<tr>
-<th>Name</th>
-<th>PPG as loser</th>
-<th>PAPG as winner</th>
-<th>Bruise Percentage</th>
-<th>Bruise Attempts per Game</th>
-<th>Rounds per Game</th>
-<th>Deuces per Round</th>
-<th>2x Deuce -> Turkey Percentage</th>
-<th>Coinflip Win Percentage</th>
-</tr>
-
-<?
-
-foreach (all_players() as $player) {
-  print "<tr>";
-
-  print "<th><a href=\"personal.php?id=$player\">";
-  print player_name($player)."</a></th>\n";
-
-  print "<td>".format_average(
-    player_overall_points_per_game_as_loser($player))."</td>\n";
-
-  print "<td>".format_average(
-    player_overall_points_against_per_game_as_winner($player))."</td>\n";
-
-  print "<td>".format_percent(
-    player_overall_bruise_percentage($player))."</td>\n";
-
-  print "<td>".format_average(
-    player_overall_bruise_attempts_per_game($player))."</td>\n";
-
-  print "<td>".format_average(
-    player_overall_rounds_per_game($player))."</td>\n";
-
-  print "<td>".format_percent(
-    player_overall_deuces_per_round($player))."</td>\n";
-
-  print "<td>".
-  format_percent(player_overall_double_deuce_to_turkey_conversion_percentage($player))."</td>";
-  print "<td>".
-  format_percent(player_overall_coinflip_win_percentage($player))."</td>";
-
-  print "</tr>\n";
-}
-
-?>
-
-</table>
-
-<br />
-
-<table width="100%" class="stats">
-<tr>
-<th>Name</th>
-<th>Total Deuces</th>
-<th>Points Scored</th>
-<th>Points Against</th>
-<th>Turkeys</th>
-<th>Tetrises</th>
-<th>Successful Bruises</th>
-<th>Missed Bruises</th>
-<th>Shutouts</th>
-<th>Been Shutout</th>
-</tr>
-
-<?
-foreach (all_players() as $player) {
-  print "<tr>";
-
-  print "<th><a href=\"personal.php?id=$player\">";
-  print player_name($player)."</a></th>\n";
-
-  print "<td>".player_total_deuces($player)."</td>\n";
-  print "<td>".player_total_points_scored($player)."</td>\n";
-  print "<td>".player_total_points_scored_against($player)."</td>\n";
-  print "<td>".player_total_turkeys($player)."</td>\n";
-  print "<td>".player_total_tetrises($player)."</td>\n";
-  print "<td>".player_total_bruises_successful($player)."</td>\n";
-  print "<td>".player_total_bruises_unsuccessful($player)."</td>\n";
-  print "<td>".player_total_shutouts($player)."</td>\n";
-  print "<td>".player_total_been_shutouts($player)."</td>\n";
-  print "</tr>\n";
-}
-
-?>
-</table>
-
-<br />
-<h2>Graphs</h2>
-<table id="graphs">
-<tr>
-<td><img src="winpercentage.php" alt="Winning Percentage Graph" /></td>
-<td><img src="deucespergame.php" alt="Deuces per Game Graph" /></td>
-</tr>
-<tr>
-<td><img src="bruisepercentage.php" alt="Bruise Percentage Graph" /></td>
-<td><img src="rollingwinpercentage.php" alt="Rolling Win Percentage Graph" /></td>
-</tr>
-</table>
-<br />
-
-<h2>Balls</h2>
-<table width="100%" class="stats">
-<tr>
-<th>Color</th>
-<th>Ball Type</th>
-<th>Use Percentage</th>
-<th>Win Percentage</th>
-<th>Deuces Per Game</th>
-</tr>
-
-<?
+$balls = Array();
 foreach (all_balls() as $ball) {
-  print "<tr>";
-
-  print "<th>".ball_color($ball)."</th>\n";
-
-  print "<td>".ball_type_name(ball_type($ball))."</td>\n";
-
-  print "<td>".
-  format_percent(ball_overall_use_percentage($ball))."</td>\n";
-
-  print "<td>".
-  format_percent(ball_overall_win_percentage($ball))."</td>\n";
-
-  print "<td>".
-  format_percent(ball_overall_deuces_per_game($ball))."</td>\n";
-
-  print "</tr>\n";
+	$balls[] = Array(
+		'color' => ball_color($ball),
+		'type' => ball_type_name(ball_type($ball)),
+		'use_perc' => format_percent(ball_overall_use_percentage($ball)),
+		'win_perc' => format_percent(ball_overall_win_percentage($ball)),
+		'deuces_per_game' => format_percent(ball_overall_deuces_per_game($ball)));
 }
 
 db_close();
-?>
-</table>
 
+$page->assign('balls', $balls);
+$page->display('statistics.tpl');
 
-</div>
-</body>
-</html>
\ No newline at end of file
+?>
Index: onegame.php
===================================================================
--- onegame.php	(revision 17)
+++ onegame.php	(working copy)
@@ -18,10 +18,7 @@
  *Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
-require("db.php");
-require("boccelib.php");
-require("header.php");
-require("side.php");
+require("start.php");
 
 $game = $_GET["game"];
 db_open();
@@ -41,75 +38,50 @@
 $winner_name = $names[0];
 $loser_name = $names[1];
 
-print "<div class=\"body\">";
-print "<h1>$winner_name vs. $loser_name</h1>";
-print "<h2>$date</h2>";
-print "<br />";
-print "<h3>Score History</h3>";
-print "<table class=\"scoreboard\">";
+$page->assign('subtitle', "$winner_name vs. $loser_name");
+$page->assign('gamedate',$date);;
+$page->assign('winner_name',$winner_name);
+$page->assign('loser_name',$loser_name);
 
-$roundtr = "<tr><th>Round</th>";
-$winnertr = "<tr><th>$winner_name</th>";
-$losertr = "<tr><th>$loser_name</th>";
-
+$scoredata = Array();
 $result = db_query("select * from point where game=$game order by round");
 while ($row = db_fetch_array($result)) {
-  $scorer = $row["scorer"];
-  $amount = $row["amount"];
-  $round = $row["round"];
+	$scorer = $row["scorer"];
+	$amount = $row["amount"];
+	$round = $row["round"];
 
-  $roundtr .= "<th>$round</th>";
-  if ($scorer == $winner) {
-    $winnertr .= "<td>$amount</td>";
-    $losertr .= "<td>0</td>";
-  } else {
-    $winnertr .= "<td>0</td>";
-    $losertr .= "<td>$amount</td>";
-  }
- }
-$roundtr .= "<th>Final</th></tr>\n";
-$winnertr .= "<td>$winner_score</td></tr>\n";
-$losertr .= "<td>$loser_score</td></tr>";
+	if ($scorer == $winner) {
+		$winner_pts = $amount;
+		$loser_pts = 0;
+	} else {
+		$winner_pts = 0;
+		$loser_pts = $amount;
+	}
+	$scoredata[] = Array(
+		'round' => $round,
+		'winner_pts' => $winner_pts,
+		'loser_pts' => $loser_pts);
+}
+$scoredata[] = Array(
+	'round' => 'Final',
+	'winner_pts' => $winner_score,
+	'loser_pts' => $loser_score);
 
-print $roundtr;
-print $winnertr;
-print $losertr;
-print "</table>";
+$page->assign('scoredata',$scoredata);
 
+$bruisedata = Array();
 $result = db_query("select * from bruise where game=$game order by round");
 if (db_num_rows($result) > 0) {
-  print '<h3>Bruises</h3>';
-  print '<table class="bruises">';
-  
-  print '<tr><th>Name</th><th>Round</th><th>Success</th></tr>';
-
-  while ($row = db_fetch_array($result)) {
-
-    //get name of bruiser
-    $name = "";
-    if ($row['player'] == $winner) {
-      $name = $winner_name;
-    } else {
-      $name = $loser_name;
-    }
-
-    print "<tr><td>$name</td>";
-
-    print "<td>" . ($row['round'] + 1) . "</td";
-
-    print "<td>";
-    if ($row['success'] == 1) {
-      print "Successful";
-    } else {
-      print "Failure";
-    }
-    print "</td>";
-  }
-  print "</table>";
+	while ($row = db_fetch_array($result))
+		$bruisedata[] = Array(
+			'name' => ($row['player']==$winner?$winner_name:$loser_name),
+			'round' => $row['round'] + 1,
+			'success' => ($row['success']?"Successful":"Failure"));
 }
+$page->assign('bruisedata', $bruisedata);
 
 db_close();
+
+$page->display('onegame.tpl');
+
 ?>
-</div>
-</body>
-</html>
\ No newline at end of file
Index: personal.php
===================================================================
--- personal.php	(revision 17)
+++ personal.php	(working copy)
@@ -18,212 +18,62 @@
  *Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
-require("db.php");
-require("boccelib.php");
+require("start.php");
 require("statslib.php");
-require("header.php");
-require("side.php");
 
 $id = $_GET["id"];
 db_open();
 
-$mainname = player_name($id);
+$page->assign('subtitle', player_name($id) . "&apos;s Statistics");
 
-?>
-<div class="body">
-<h1><?=$mainname?>&apos;s Statistics</h1>
-<div style="padding-top: 5px; padding-bottom: 15px; font-size: small;">
-<a href="statistics.php">Main Statistics</a>
-</div>
-<table width="100%" class="stats">
-<tr>
-<th>Name</th>
-<th>Record</th>
-<th>Win Percentage</th>
-<th>Deuces Per Game</th>
-<th>Deuces Against Per Game</th>
-<th>Points Per Game</th>
-<th>Points Against Per Game</th>
-<th>Current Streak</th>
-</tr>
-<?
-
+$players = Array();
 foreach (all_players() as $player) {
-  if ($player == $id) {
-    continue;
-  }
+	if ($player == $id)
+		continue;
 
-  print "<tr>";
-
-  print "<th><a href=\"personal.php?id=$player\">";
-  print player_name($player)."</a></th>\n";
-  
-  print "<td>".player_total_wins_versus($id, $player)."&nbsp;-&nbsp;";
-  print player_total_losses_versus($id, $player)."</td>\n";
-  
-  print "<td>".
-  format_percent(player_overall_win_percentage_versus($id, $player)).
-  "</td>\n";
-  
-  print "<td>".
-  format_average(player_overall_deuces_per_game_versus($id, $player)).
-  "</td>\n";
-
-  print "<td>".
-  format_average(player_overall_deuces_against_per_game_versus($id,
-						       $player))."</td>\n";
-
-  print "<td>".
-  format_average(player_overall_points_scored_per_game_versus($id,
-						       $player))."</td>\n";
-
-  print "<td>".
-  format_average(player_overall_points_scored_against_per_game_versus($id,
-						       $player))."</td>\n";
-
-  print "<td>".player_overall_current_streak_versus($id, $player)."</td>\n";
-
-  print "</tr>\n";
+	$players[] = Array(
+		'id' => $player,
+		'name' => player_name($player),
+ 		'wins' => player_total_wins_versus($id, $player),
+		'loses' => player_total_losses_versus($id, $player),
+		'win_perc' => format_percent(player_overall_win_percentage_versus($id, $player)),
+		'deuces' => format_average(player_overall_deuces_per_game_versus($id, $player)),
+		'deuces_against' => format_average(player_overall_deuces_against_per_game_versus($id,$player)),
+		'points_per_game' => format_average(player_overall_points_scored_per_game_versus($id,$player)),
+		'points_against_per_game' => format_average(player_overall_points_scored_against_per_game_versus($id,$player)),
+		'streak' => player_overall_current_streak_versus($id, $player),
+		// table 2
+		'bruise_perc' => format_percent(player_overall_bruise_percentage_versus($id, $player)),
+		'bruise_att_pg' => format_average(player_overall_bruises_attempted_per_game_versus($id, $player)),
+		'rounds_pg' => format_average(player_overall_rounds_per_game_versus($id, $player)),
+		'deuces_pr' => format_percent(player_overall_deuces_per_round_versus($id, $player)),
+		'coin_wins' => format_percent(player_overall_coinflip_win_percentage_versus($id, $player)),
+		// table 3
+		'pts' => player_total_points_scored_versus($id, $player),
+		'pts_against' => player_total_points_scored_against_versus($id,$player),
+		'bruise_att' => player_total_bruises_attempted_versus($id, $player),
+		'turkeys' => player_total_turkeys_versus($id, $player),
+		'shutouts' => player_total_shutouts_versus($id, $player),
+		'been_shutouts' => player_total_been_shutouts_versus($id, $player),
+		'games_perc' => format_percent(player_total_games_played_percent_versus($id, $player)));
 }
 
-?>
-</table>
+$page->assign('players',$players);
 
-<br />
-
-<table width="100%" class="stats">
-<tr>
-<th>Name</th>
-<th>Bruise Percentage</th>
-<th>Bruises Per Game</th>
-<th>Rounds per Game</th>
-<th>Deuces per Round</th>
-<th>Coin Flip Win Percentage</th>
-</tr>
-<?
-foreach (all_players() as $player) {
-  if ($player == $id) {
-    continue;
-  }
-
-  print "<tr>";
-
-  print "<th><a href=\"personal.php?id=$player\">";
-  print player_name($player)."</a></th>\n";
-
-  print "<td>".
-  format_percent(player_overall_bruise_percentage_versus($id, $player)).
-  "</td>\n";
-
-  print "<td>".
-  format_average(player_overall_bruises_attempted_per_game_versus($id,
-							$player))."</td>\n";
-
-  print "<td>".
-  format_average(player_overall_rounds_per_game_versus($id, $player)).
-  "</td>\n";
-
-  print "<td>".
-  format_percent(player_overall_deuces_per_round_versus($id, $player)).
-  "</td>\n";
-
-  print "<td>".
-  format_percent(player_overall_coinflip_win_percentage_versus($id, $player)).
-  "</td>\n";
-
-  print "</tr>\n";
-}
-
-?>
-</table>
-<br />
-
-<table width="100%" class="stats">
-<tr>
-<th>Name</th>
-<th>Points Scored</th>
-<th>Points Scored Against</th>
-<th>Bruises</th>
-<th>Turkeys</th>
-<th>Shutouts</th>
-<th>Been Shutout</th>
-<th>Percent Games Played</th>
-</tr>
-
-<?
-foreach (all_players() as $player) {
-  if ($player == $id) {
-    continue;
-  }
-
-  print "<tr>";
-
-  print "<th><a href=\"personal.php?id=$player\">";
-  print player_name($player)."</a></th>\n";
-
-  print "<td>".player_total_points_scored_versus($id, $player)."</td>\n";
-
-  print "<td>".player_total_points_scored_against_versus($id,$player).
-  "</td>\n";
-
-  print "<td>".player_total_bruises_attempted_versus($id, $player).
-  "</td>\n";
-
-  print "<td>".player_total_turkeys_versus($id, $player)."</td>\n";
-
-  print "<td>".player_total_shutouts_versus($id, $player)."</td>\n";
-
-  print "<td>".player_total_been_shutouts_versus($id, $player)."</td>\n";
-
-  print "<td>".
-  format_percent(player_total_games_played_percent_versus($id, $player)).
-  "</td>\n";
-
-  print "</tr>\n";
-}
-
-?>
-
-</table>
-
-<br />
-
-<h2>Balls</h2>
-<table width="100%" class="stats">
-<tr>
-<th>Color</th>
-<th>Type</th>
-<th>Use Percentage</th>
-<th>Win Percentage</th>
-<th>Deuces Per Game</th>
-</tr>
-<?
+$balls = Array();
 foreach (all_balls() as $ball) {
-  print "<tr>";
-
-  print "<th>".ball_color($ball)."</th>\n";
-  print "<td>".ball_type_name(ball_type($ball))."</td>\n";
-
-  print "<td>".
-  format_percent(player_overall_ball_use_percentage($id, $ball)).
-  "</td>\n";
-
-  print "<td>".format_percent(player_overall_win_percentage_with_ball($id,
-								      $ball)).
-  "</td>\n";
-
-  print "<td>".
-  format_percent(player_overall_deuces_per_game_with_ball($id, $ball)).
-  "</td>\n";
-
-  print "</tr>\n";
+	$balls[] = Array(
+		'color' => ball_color($ball),
+		'type' => ball_type_name(ball_type($ball)),
+		'use_perc' => format_percent(player_overall_ball_use_percentage($id, $ball)),
+		'win_perc' => format_percent(player_overall_win_percentage_with_ball($id,$ball)),
+		'deuces_pg' => format_percent(player_overall_deuces_per_game_with_ball($id, $ball)));
 }
 
 db_close();
 
-?>
-</table>
+$page->assign('balls', $balls);
 
-</div>
-</body>
-</html>
\ No newline at end of file
+$page->display('personal.tpl');
+
+?>
Index: allscores.php
===================================================================
--- allscores.php	(revision 17)
+++ allscores.php	(working copy)
@@ -18,16 +18,7 @@
  *Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
-require("db.php");
-require("boccelib.php");
-require("header.php");
-require("side.php");
-?>
-<div class="body">
-<h1>Office Bocce Games</h1>
-<table class="allgames">
-<tr><th>Winner</th><th>Loser</th><th>Score</th><th>Date & Time</th></tr>
-<?
+require("start.php");
 
 db_open();
 $result = db_query("select * from player");
@@ -38,6 +29,8 @@
   $players[$row["id"]] = $row["firstname"];
  }
 
+$data = Array();
+
 $result = db_query("select id,winner,loser,winner_points,loser_points,strftime('%s',date) as date from game order by date desc");
 while ($row = db_fetch_array($result)) {
   $winner_name = $players[$row["winner"]];
@@ -45,11 +38,17 @@
   $score = $row["winner_points"] . " - " . $row["loser_points"];
   $date = date("M jS, Y @ g:i a",$row["date"]);
   $id = $row["id"];
-  echo "<tr><td>$winner_name</td><td>$loser_name</td><td>$score</td><td><a href='onegame.php?game=$id'>$date</a></td></tr>";
+  $data[] = Array(
+  		'id' => $id,
+  		'winner_name' => $winner_name,
+		'loser_name' => $loser_name,
+		'score' => $score,
+		'date' => $date);
  }
 db_close();
+
+$page->assign('subtitle', 'Office Bocce Games');
+$page->assign('data', $data);
+$page->display('allscores.tpl');
+
 ?>
-</table>
-</div>
-</body>
-</html>
\ No newline at end of file
Index: start.php
===================================================================
--- start.php	(revision 0)
+++ start.php	(revision 0)
@@ -0,0 +1,48 @@
+<?
+/*
+ * Copyright (C) 2005 Josef "Jeff" Sipek <jeffpc@optonline.net>
+ *
+ *This program is free software; you can redistribute it and/or modify
+ *it under the terms of the GNU General Public License as published by
+ *the Free Software Foundation; either version 2 of the License, or
+ *(at your option) any later version.
+ *
+ *This program is distributed in the hope that it will be useful,
+ *but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *GNU General Public License for more details.
+ *
+ *You should have received a copy of the GNU General Public License
+ *along with this program; if not, write to the Free Software
+ *Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+require_once("config.php");
+
+/*
+ * Start up the session
+ */
+session_start();
+
+/*
+ * Set up Smarty
+ */
+require_once(SMARTY_DIR."Smarty.class.php");
+
+$page = new Smarty;
+$page->template_dir	= $template_dir;
+$page->compile_dir	= $compile_dir;
+$page->config_dir	= $config_dir;
+$page->cache_dir	= $cache_dir;
+
+/*
+ * Set up db
+ */
+require_once("db.php");
+
+/*
+ * Start up boccelib
+ */
+require("boccelib.php");
+
+?>
Index: flipcoin.php
===================================================================
--- flipcoin.php	(revision 17)
+++ flipcoin.php	(working copy)
@@ -17,166 +17,125 @@
  *Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
-session_start();
-require('boccelib.php');
-  db_open();
+require('start.php');
 
-if (!isset($_POST["submit"])) {
-  coinFlipResults();
+db_open();
 
+if (!isset($_POST["submit"])) {
+	coinFlipResults();
+	$template = 'results';
 } elseif ($_POST["submit"] == "Go First") {
-  chooseGoFirst();
-
+	chooseGoFirst();
+	$template = 'gofirst';
 } elseif ($_POST["submit"] == "Choose Balls") {
-  chooseBallColor();
+	chooseBallColor();
+	$template = 'ballcolor';
 }
 
+$page->display("flipcoin-$template.tpl");
+exit(0);
+
 function chooseGoFirst() {
-  require("header.php");
-  require("side.php");
+	global $page;
 
-  //if the winner of the coin toss selected to 'go first'
-  $_SESSION["pallino_toss"] = array();
-  $_SESSION["pallino_toss"][0] = $_SESSION["acting_player"];
-  $chose_first_name = player_name($_SESSION["acting_player"]);
+	//if the winner of the coin toss selected to 'go first'
+	$_SESSION["pallino_toss"] = array();
+	$_SESSION["pallino_toss"][0] = $_SESSION["acting_player"];
+	$chose_first_name = player_name($_SESSION["acting_player"]);
 
-  $_SESSION["acting_player"] = nonActingPlayer();
-  $acting_player_name = player_name($_SESSION["acting_player"]);
+	$_SESSION["acting_player"] = nonActingPlayer();
+	$acting_player_name = player_name($_SESSION["acting_player"]);
 
-  ?>
-  <div class="body">
-  <form action="" method="post">
-    <h1 class="title"><?=$acting_player_name?>, Choose Your Balls</h1>
-    <div style="padding-top: 1em; width: 12em; text-align: right;">
-      <?=$chose_first_name?> chose to go first.
-    </div>
-    <form action="" method="post">
-    <div class="labels">
-      <? select_balls("Choose your balls:"); ?>
-    </div>
-  </form>
-  </div>
-  <?
+	$page->assign('subtitle', "$acting_player_name, Choose Your Balls");
 
-  db_close();
+	$page->assign('chose_first_name', $chose_first_name);
+        select_balls("Choose your balls:");
+	
+	db_close();
 }
 
 function chooseBallColor() {
-  require("header.php");
-  require("side.php");
+	global $page;
 
-  if ($_SESSION["acting_player"] == $_SESSION["player1"]) {
-    $player = "player1";
-  } else {
-    $player = "player2";
-  }
+	if ($_SESSION["acting_player"] == $_SESSION["player1"]) {
+		$player = "player1";
+	} else {
+		$player = "player2";
+	}
 
-  $_SESSION[$player."_ball1"]=$_POST['ball1'];
-  $_SESSION[$player."_ball2"]=$_POST['ball2'];
+	$_SESSION[$player."_ball1"]=$_POST['ball1'];
+	$_SESSION[$player."_ball2"]=$_POST['ball2'];
 
-  if (!isset($_SESSION["pallino_toss"][0])) {
-    $_SESSION["pallino_toss"][0] = nonActingPlayer();
-  }
+	if (!isset($_SESSION["pallino_toss"][0])) {
+		$_SESSION["pallino_toss"][0] = nonActingPlayer();
+	}
 
-  //if we're done
-  if (isset($_SESSION["player1_ball1"]) &&
-      isset($_SESSION["player1_ball2"]) &&
-      isset($_SESSION["player2_ball1"]) &&
-      isset($_SESSION["player2_ball2"]) &&
-      isset($_SESSION["pallino_toss"][0])) {
+	//if we're done
+	if (isset($_SESSION["player1_ball1"]) &&
+	    isset($_SESSION["player1_ball2"]) &&
+	    isset($_SESSION["player2_ball1"]) &&
+	    isset($_SESSION["player2_ball2"]) &&
+	    isset($_SESSION["pallino_toss"][0])) {
+		unset($_SESSION["acting_player"]);
 
-    unset($_SESSION["acting_player"]);
+		redirect("addscore.php");
+		db_close();
+		return 0;
+	}
 
-    redirect("addscore.php");
-    db_close();
-    return 0;
-  }
+	$chose_balls_name = player_name($_SESSION["acting_player"]);
+	$page->assign('chose_balls_name', $chose_balls_name);
 
-  $chose_balls_name = player_name($_SESSION["acting_player"]);
+	$_SESSION["acting_player"] = nonActingPlayer();
+	$acting_player_name = player_name($_SESSION["acting_player"]);
 
-  $_SESSION["acting_player"] = nonActingPlayer();
-  $acting_player_name = player_name($_SESSION["acting_player"]);
+	$page->assign('subtitle', "$acting_player_name, Choose Your Balls");
 
-  $big_ball_name = ball_type_name($GLOBALS["BRUISER_TYPE"]);
-  $small_ball_name = ball_type_name($GLOBALS["LITTLE_BALL_TYPE"]);
+	$prev_balls = Array(
+		Array(
+			'name' => ball_type_name($GLOBALS["BRUISER_TYPE"]),
+			'color' => ball_color($_POST['ball1'])
+		),
+		Array(
+			'name' => ball_type_name($GLOBALS["LITTLE_BALL_TYPE"]),
+			'color' => ball_color($_POST['ball2'])));
+	$page->assign('prev_balls', $prev_balls);
 
-  $big_ball_color = ball_color($_POST['ball1']);
-  $small_ball_color = ball_color($_POST['ball2']);
+	select_balls("Choose your balls:");
 
-  ?>
-  <div class="body">
-  <form action="" method="post">
-    <h1 class="title"><?=$acting_player_name?>, Choose Your Balls</h1>
-    <div style="padding-top: 1em; padding-left: 1em;">
-      <?=$chose_balls_name?> chose to use the following balls:
-    </div>
-    <div style="padding-left: 6em;">
-      <?=$big_ball_name?>: <?=$big_ball_color?><br />
-      <?=$small_ball_name?>: <?=$small_ball_color?>
-    </div>
-    <form action="" method="post">
-    <div class="labels">
-      <? select_balls("Choose your balls:"); ?>
-    </div>
-  </form>
-  </div>
-  <?
-
-  db_close();
+	db_close();
 }
 
 function coinFlipResults() {
-  require("header.php");
-  require("side.php");
+	global $page;
+	
+	$coinflip_winner = (flip_coin() + 1);
 
-  $coinflip_winner = (flip_coin() + 1);
+	//the acting_player is the player that is making the current decision
+	$coinflip_winner = "player" . $coinflip_winner;
+	$_SESSION["acting_player"] = $_SESSION[$coinflip_winner];
+	$_SESSION["coinflip_winner"] = $_SESSION[$coinflip_winner];
 
-  //the acting_player is the player that is making the current decision
-  $coinflip_winner = "player" . $coinflip_winner;
-  $_SESSION["acting_player"] = $_SESSION[$coinflip_winner];
-  $_SESSION["coinflip_winner"] = $_SESSION[$coinflip_winner];
-  $coinflip_winner_name = player_name($_SESSION["coinflip_winner"]);
+	$page->assign('subtitle', player_name($_SESSION["coinflip_winner"]) . " Won the Coin Toss!");
   
-  ?>
-  <div class="body">
-  <form action="" method="post">
-  <h1 class="title"><?=$coinflip_winner_name?> Won the Coin Toss!</h1>
+	select_balls("I want to choose my balls:");
   
-  <form action="" method="post">
-    <div class="labels">
-      <div class="label" style="padding-bottom: 10px;">
-        <div style="width: 12em;">I want to go first:</div>
-        <input name="submit" type="submit" value="Go First" />
-      </div>
-     <? select_balls("I want to choose my balls:"); ?>
-    </div>
-  </form>
-  </div>
-  <?
-
-  db_close();
+	db_close();
 }
   
 function select_balls($message) {
-  $bruiserselect = get_ball_select($GLOBALS['BRUISER_TYPE']);
-  $littleballselect = get_ball_select($GLOBALS['LITTLE_BALL_TYPE']);
+	global $page;
 
-?>
-  <div class="label">
-    <div style="width: 12em;"><?=$message?></div>
-    <?=$bruiserselect?>
-    <?=$littleballselect?>
-    <br />
-    <div style="text-align: right; width: 25.7em; padding-top: 0px;">
-      <input name="submit" type="submit" value="Choose Balls" />
-    </div>
-  </div>
-<?
+	$bruisers = get_balls($GLOBALS['BRUISER_TYPE']);
+	$littleballs = get_balls($GLOBALS['LITTLE_BALL_TYPE']);
+
+	$page->assign('balls_message', $message);
+	$page->assign('balls', Array($bruisers, $littleballs));
 }
 
 function flip_coin() {
-  //this function just returns a 0 or a 1
-  return (mt_rand() & 1);
+	//this function just returns a 0 or a 1
+	return (mt_rand() & 1);
 }
 
 function nonActingPlayer() {
@@ -188,38 +147,34 @@
   }
 }
 
-function get_ball_select($type) {
-  $query = "select id, color from ball where num=$type order by color";
-  $result = db_query($query);
+function get_balls($type) {
+	$query = "select id, color from ball where num=$type order by color";
+	$result = db_query($query);
 
-  $balls = array();
+	$balls = array();
 
-  while($row = db_fetch_array($result)) {
-    $balls[$row['id']] = $row['color'];
-  }
+	while($row = db_fetch_array($result)) {
+		$balls[$row['id']] = $row['color'];
+	}
 
-  $ball_type_name = ball_type_name($type);
+	$ball_type_name = ball_type_name($type);
 
-  $exclude = "";
-  for ($i = 1; $i <= 2; $i++) {
-    if (isset($_SESSION["player$i" . "_ball$type"])) {
-      $exclude = $_SESSION["player$i" . "_ball$type"];
-    }
-  }
+	$exclude = "";
+	for ($i = 1; $i <= 2; $i++) {
+		if (isset($_SESSION["player$i" . "_ball$type"])) {
+			$exclude = $_SESSION["player$i" . "_ball$type"];
+		}
+	}
 
-  $ballselect = "<select name=\"ball$type\">\n";
-  $ballselect .= "<option value=\"-----\">$ball_type_name</option>\n";
-  foreach ($balls as $id => $ball) {
-    if ($id != $exclude) {
-      $ballselect .= "<option value=\"$id\">$ball</option>\n";
-    }
-  }
-  $ballselect .= "</select>\n";
+	$balldata = Array();
+	foreach ($balls as $id => $ball) {
+		if ($id != $exclude) {
+			$balldata[] = Array('id' => $id, 'name' => $ball);
+		}
+	}
   
-  return $ballselect;
+	return Array('type' => $type, 'type_name' => $ball_type_name, 'balls' => $balldata);
 }
 
-?>
 
-</body>
-</html>
\ No newline at end of file
+?>
Index: addplayer.php
===================================================================
--- addplayer.php	(revision 17)
+++ addplayer.php	(working copy)
@@ -18,8 +18,7 @@
  *Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
-require ('db.php');
-require ('boccelib.php');
+require ('start.php');
 
 $notice = "";
 
@@ -93,48 +92,9 @@
   }
 }
 
-require("header.php");
-require("side.php");
+$page->assign('notice', $notice);
+$page->assign('subtitle', "Add a Bocce Player");
 
-print "<div class=\"body\">";
+$page->display('addplayer.tpl');
 
-if ($notice != "") {
-  print "<div class=\"notice\">$notice</div>";
-}
 ?>
-<h1>Add a Bocce Player</h1>
-
-<form action="" method="post">
-  <div class="labels">
-    <div class="label">
-      <div>First Name:</div>
-      <input type="text" maxlength="25" size="25" name="fname" value="<?=$fname?>" />
-    </div>
-    <div class="label">
-      <div>Last Name:</div>
-      <input type="text" maxlength="25" size="25" name="lname" value="<?=$lname?>" />
-    </div>
-    <div class="label">
-      <div>User Name:</div>
-      <input type="text" maxlength="16" size="25" name="uname" value="<?=$uname?>" />
-    </div>
-    <div class="label">
-      <div>Email Address:</div>
-      <input type="text" size="25" name="email" value="<?=$email?>" />
-    </div>
-    <div class="label" id="password">
-      <div>Password:</div>
-      <input type="password" maxlength="25" size="25" name="pass" />
-    </div>
-    <div class="label">
-      <div>Verify Password:</div>
-      <input type="password" maxlength="25" size="25" name="vpass" />
-    </div>
-    <div class="submit">
-      <input class="submit" name="submit" type="submit" value="Submit" />
-    </div>
-  </div>
-</form>
-</div>
-</body>
-</html>
Index: config.php
===================================================================
--- config.php	(revision 17)
+++ config.php	(working copy)
@@ -1,6 +1,15 @@
 <?
 $jpgraph_dir = "../jpgraph-1.17beta2";
 
+# Location of smarty class
+define(SMARTY_DIR,'/usr/share/php/smarty/libs/');
+
+# Smarty directories
+$template_dir	= 'templates';
+$compile_dir	= 'templates_c';
+$config_dir	= 'configs';
+$cache_dir	= 'cache';
+
 # Database engine to use
 #    Choices:
 #	sqlite
Index: index.php
===================================================================
--- index.php	(revision 17)
+++ index.php	(working copy)
@@ -18,17 +18,8 @@
  *Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
-print "<?xml version=\"1.0\" encoding=\"UTF-8\"?>";
-print "<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.1//EN\" \"http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd\">";
-require("boccelib.php");
-require("header.php");
-require("side.php");
+require("start.php");
 
+$page->display('index.tpl');
+
 ?>
-<div id="mainbody">
-<p>This is meant to be an exercise in web development with php and sqlite.  This is also a great excuse to go extrememly overboard with statistics for bocce, but it should be trivial adding support for other database engines.</p>
-<img src="images/pope.gif" alt="The pope playing bocce" />
-</div>
-<div class="cleaner" />
-</body>
-</html>
\ No newline at end of file
Index: flipping.php
===================================================================
--- flipping.php	(revision 17)
+++ flipping.php	(working copy)
@@ -17,20 +17,11 @@
  *Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
-require("header.php");
-require("side.php");
-?>
+require("start.php");
 
-<div class="body">
-<h1>Flipping the Coin ...</h1>
-</div>
+$page->assign('subtitle', "Flipping the coin");
 
-<script>
-function gotoFlipCoin() {
-  window.location="flipcoin.php";
-}
-setTimeout("gotoFlipCoin()", 2500);
-</script>
-</div>
-</body>
-</html>
\ No newline at end of file
+$page->display('flipping.tpl');
+
+?>
+
Index: templates/personal.tpl
===================================================================
--- templates/personal.tpl	(revision 0)
+++ templates/personal.tpl	(revision 0)
@@ -0,0 +1,118 @@
+{include file="header.tpl" title=$title}
+
+<div style="padding-top: 5px; padding-bottom: 15px; font-size: small;">
+<a href="statistics.php">Main Statistics</a>
+</div>
+<table width="100%" class="stats">
+<tr>
+<th>Name</th>
+<th>Record</th>
+<th>Win Percentage</th>
+<th>Deuces Per Game</th>
+<th>Deuces Against Per Game</th>
+<th>Points Per Game</th>
+<th>Points Against Per Game</th>
+<th>Current Streak</th>
+</tr>
+
+{section name=mysec loop=$players}
+{strip}
+<tr>
+<th><a href="personal.php?id={$players[mysec].id}">{$players[mysec].name}</a></th>
+<td>{$players[mysec].wins}&nbsp;-&nbsp;{$players[mysec].loses}</td>
+<td>{$players[mysec].win_perc}</td>
+<td>{$players[mysec].deuces}</td>
+<td>{$players[mysec].deuces_against}</td>
+<td>{$players[mysec].points_per_game}</td>
+<td>{$players[mysec].points_against_per_game}</td>
+<td>{$players[mysec].streak}</td>
+</tr>
+{/strip}
+{/section}
+
+</table>
+
+<br />
+
+<table width="100%" class="stats">
+<tr>
+<th>Name</th>
+<th>Bruise Percentage</th>
+<th>Bruises Per Game</th>
+<th>Rounds per Game</th>
+<th>Deuces per Round</th>
+<th>Coin Flip Win Percentage</th>
+</tr>
+
+{section name=mysec loop=$players}
+{strip}
+<tr>
+<th><a href="personal.php?id={$players[mysec].id}">{$players[mysec].name}</a></th>
+<td>{$players[mysec].bruise_perc}</td>
+<td>{$players[mysec].bruise_att_pg}</td>
+<td>{$players[mysec].rounds_pg}</td>
+<td>{$players[mysec].deuces_pr}</td>
+<td>{$players[mysec].coin_wins}</td>
+</tr>
+{/strip}
+{/section}
+
+</table>
+<br />
+
+<table width="100%" class="stats">
+<tr>
+<th>Name</th>
+<th>Points Scored</th>
+<th>Points Scored Against</th>
+<th>Bruises</th>
+<th>Turkeys</th>
+<th>Shutouts</th>
+<th>Been Shutout</th>
+<th>Percent Games Played</th>
+</tr>
+
+{section name=mysec loop=$players}
+{strip}
+<tr>
+<th><a href="personal.php?id={$players[mysec].id}">{$players[mysec].name}</a></th>
+<td>{$players[mysec].pts}
+<td>{$players[mysec].pts_against}
+<td>{$players[mysec].bruise_att}
+<td>{$players[mysec].turkeys}
+<td>{$players[mysec].shutouts}
+<td>{$players[mysec].been_shutouts}
+<td>{$players[mysec].games_perc}
+</tr>
+{/strip}
+{/section}
+
+</table>
+
+<br />
+
+<h2>Balls</h2>
+<table width="100%" class="stats">
+<tr>
+<th>Color</th>
+<th>Type</th>
+<th>Use Percentage</th>
+<th>Win Percentage</th>
+<th>Deuces Per Game</th>
+</tr>
+
+{section name=mysec loop=$balls}
+{strip}
+<tr>
+<th>{$balls[mysec].color}</th>
+<td>{$balls[mysec].type}</td>
+<td>{$balls[mysec].use_perc}</td>
+<td>{$balls[mysec].win_perc}</td>
+<td>{$balls[mysec].deuces_pg}</td>
+</tr>
+{/strip}
+{/section}
+
+</table>
+
+{include file="footer.tpl"}
Index: templates/addscore-progress.tpl
===================================================================
--- templates/addscore-progress.tpl	(revision 0)
+++ templates/addscore-progress.tpl	(revision 0)
@@ -0,0 +1,91 @@
+{include file="header.tpl" title=$title}
+
+<form action="" method="post">
+<center>
+<table class="score">
+<tr>
+<td><h1>{$score1}</h1></td>
+<td colspan="2"><h1>-</h1></td>
+<td><h1>{$score2}</h1></td>
+</tr>
+<tr>
+<td><h2>{$name1}</h2></td>
+<td colspan="2"></td>
+<td><h2>{$name2}</h2></td>
+</tr>
+</table>
+<br />
+<table class="controls">
+{if !$confirmPage}
+<tr><td style="border: 0px;">
+<table class="controlpad">
+<tr>
+<td>{$name1}</td>
+<td><button type="submit" name="action" value="Player1Score1">Score 1</button></td>
+<td><button type="submit" name="action" value="Player1Score2" >Deuce</button></td>
+<td><button type="submit" name="action" value="Player1GoodBruise" >Successful Bruise</button></td>
+<td><button type="submit" name="action" value="Player1BadBruise" >Missed Bruise</button></td>
+</tr>
+<tr>
+<td>{$name2}</td>
+<td><button type="submit" name="action" value="Player2Score1">Score 1</button></td>
+<td><button type="submit" name="action" value="Player2Score2" >Deuce</button></td>
+<td><button type="submit" name="action" value="Player2GoodBruise" >Successful Bruise</button></td>
+<td><button type="submit" name="action" value="Player2BadBruise" >Missed Bruise</button></td>
+</tr>
+</table>
+</td>
+</tr>
+{/if}
+
+{if $undo}
+<tr><td colspan="5" style="border: 0px"><button type="submit" name="action" value="undo">Undo</button></td></tr>
+{/if}
+
+{if $confirmPage}
+<tr><td colspan="5" style="border: 0px;"><button type="submit" name="action" value="confirm">Confirm</button></td></tr>
+{/if}
+</table>
+
+{if $showhistory}
+<h3>Score History</h3>
+<table class="scoreboard">
+
+<tr>
+<td>{$name1}</td>
+{section name=hist loop=$hist1}
+<td>{$hist1[hist]}</td>
+{/section}
+</tr>
+
+<tr>
+<td>{$name2}</td>
+{section name=hist loop=$hist2}
+<td>{$hist2[hist]}</td>
+{/section}
+</tr>
+</table>
+{/if}
+
+{if $showbruises}
+<h3>Bruises</h3>
+<table class="bruises">
+<tr>
+<th>Name</th><th>Round</th><th>Success</th>
+</tr>
+{section name=br loop=$bruises}
+{strip}
+<tr>
+<td>{$bruises[br].name}</td>
+<td>{$bruises[br].round}</td>
+<td>{$bruises[br].succ}</td>
+</tr>
+{/strip}
+{/section}
+</table>
+{/if}
+
+</center>
+</form>
+
+{include file="footer.tpl"}
Index: templates/startgame.tpl
===================================================================
--- templates/startgame.tpl	(revision 0)
+++ templates/startgame.tpl	(revision 0)
@@ -0,0 +1,34 @@
+<!-- {$subtitlehint} -->
+{include file="header.tpl" title=$title}
+
+{if $notice}
+<div id="notice">{$notice}</div>
+{/if}
+<form action="" method="post">
+  <div class="labels">
+    <div class="label">
+      <div>User Name:</div>
+      <input type="text" maxlength="16" size="25" name="uname" value="{$uname}" />
+    </div>
+    <div class="label">
+      <div>Password:</div>
+      <input type="password" maxlength="25" size="25" name="pass" />
+    </div>
+    <div class="submit">
+      <input name="submit" type="submit" value="Login" />
+      <input name="submit" type="submit" value="Cancel" />
+    </div>
+  </div>
+</form>
+</div>
+
+{literal}
+<script type="text/javascript">
+function setfocus() {
+  document.forms[0].uname.focus();
+}
+window.onload=setfocus;
+</script>
+{/literal}
+
+{include file="footer.tpl"}
Index: templates/allscores.tpl
===================================================================
--- templates/allscores.tpl	(revision 0)
+++ templates/allscores.tpl	(revision 0)
@@ -0,0 +1,12 @@
+{include file="header.tpl" title=$title}
+
+<table class="allgames">
+<tr><th>Winner</th><th>Loser</th><th>Score</th><th>Date &amp; Time</th></tr>
+{section name=mysec loop=$data}
+{strip}
+<tr><td>{$data[mysec].winner_name}</td><td>{$data[mysec].loser_name}</td><td>{$data[mysec].score}</td><td><a href='onegame.php?game={$data[mysec].id}'>{$data[mysec].date}</a></td></tr>
+{/strip}
+{/section}
+</table>
+
+{include file="footer.tpl"}
Index: templates/header.tpl
===================================================================
--- templates/header.tpl	(revision 0)
+++ templates/header.tpl	(revision 0)
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN" "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">
+<html>
+<head>
+<title>OfficeBocce.com</title>
+<link rel="stylesheet" type="text/css" href="main.css" />
+</head>
+<body>
+<div id="wrapper">
+<div id="header">
+<a href="index.php"><img src="images/bocceball.png" alt="Bocce Ball" /></a>
+<span id="title">{$title|default:"Office Bocce Score Tracker"}</span>
+</div> <!--header-->
+
+{include file="sidepanel.tpl"}
+
+<div id="content">
+{if $subtitle != ""}
+<div id="subtitle">
+{strip}
+<h1>{$subtitle}</h1>
+{if $subtitlehint != ""}
+<div id="subtitlehint">{$subtitlehint}</div>
+{/if}
+{/strip}
+</div>
+{/if}
+<!-- content begins -->
Index: templates/statistics.tpl
===================================================================
--- templates/statistics.tpl	(revision 0)
+++ templates/statistics.tpl	(revision 0)
@@ -0,0 +1,130 @@
+{include file="header.tpl" title=$title}
+
+<br />
+<table width="100%" class="stats">
+<tr>
+<th>Name</th>
+<th>Record (W-L)</th>
+<th>Win Percentage</th>
+<th>Deuces Per Game</th>
+<th>Deuces Against Per Game</th>
+<th>Points per Game</th>
+<th>PA per Game</th>
+<th>Current Streak</th>
+</tr>
+
+{section name=mysec loop=$players}
+{strip}
+<tr>
+<th><a href="personal.php?id={$players[mysec].id}">{$players[mysec].name}</a></th>
+<td>{$players[mysec].wins}&nbsp;-&nbsp;{$players[mysec].loses}</td>
+<td>{$players[mysec].overall_win_perc}</td>
+<td>{$players[mysec].overall_deuces_per_game}</td>
+<td>{$players[mysec].overall_deuces_against_per_game}</td>
+<td>{$players[mysec].overall_points_per_game}</td>
+<td>{$players[mysec].overall_points_against_per_game}</td>
+<td>{$players[mysec].overall_current_streak}</td>
+</tr>
+{/strip}
+{/section}
+</table>
+
+<br />
+
+<table width="100%" class="stats">
+<tr>
+<th>Name</th>
+<th>PPG as loser</th>
+<th>PAPG as winner</th>
+<th>Bruise Percentage</th>
+<th>Bruise Attempts per Game</th>
+<th>Rounds per Game</th>
+<th>Deuces per Round</th>
+<th>2x Deuce -&gt; Turkey Percentage</th>
+<th>Coinflip Win Percentage</th>
+</tr>
+
+{section name=mysec loop=$players}
+{strip}
+<tr>
+<th><a href="personal.php?id={$players[mysec].id}">{$players[mysec].name}</a></th>
+<td>{$players[mysec].overall_points_per_game_as_loser}</td>
+<td>{$players[mysec].overall_points_against_per_game_as_winner}</td>
+<td>{$players[mysec].overall_bruise_perc}</td>
+<td>{$players[mysec].overall_bruise_attempts_per_game}</td>
+<td>{$players[mysec].overall_rounds_per_game}</td>
+<td>{$players[mysec].overall_deuces_per_round}</td>
+<td>{$players[mysec].overall_double_deuce_to_turkey_conv_perc}</td>
+<td>{$players[mysec].overall_coinflip_win_perc}</td>
+</tr>
+{/strip}
+{/section}
+</table>
+
+<br />
+
+<table width="100%" class="stats">
+<tr>
+<th>Name</th>
+<th>Total Deuces</th>
+<th>Points Scored</th>
+<th>Points Against</th>
+<th>Turkeys</th>
+<th>Tetrises</th>
+<th>Successful Bruises</th>
+<th>Missed Bruises</th>
+<th>Shutouts</th>
+<th>Been Shutout</th>
+</tr>
+
+{section name=mysec loop=$players}
+{strip}
+<tr>
+<th><a href="personal.php?id={$players[mysec].id}">{$players[mysec].name}</a></th>
+<td>{$players[mysec].total_deuces}</td>
+<td>{$players[mysec].total_points_scored}</td>
+<td>{$players[mysec].total_points_scored_against}</td>
+<td>{$players[mysec].total_turkeys}</td>
+<td>{$players[mysec].total_tetrises}</td>
+<td>{$players[mysec].total_bruises_succ}</td>
+<td>{$players[mysec].total_bruises_unsecc}</td>
+<td>{$players[mysec].total_shutouts}</td>
+<td>{$players[mysec].total_been_shutouts}</td>
+</tr>
+{/strip}
+{/section}
+</table>
+
+<br />
+<h2>Graphs</h2>
+<img src="winpercentage.php" alt="Winning Percentage Graph" class="leftimage" />
+<img src="deucespergame.php" alt="Deuces per Game Graph" class="rightimage" />
+<br/>
+<img src="bruisepercentage.php" alt="Bruise Percentage Graph" class="leftimage" />
+<img src="rollingwinpercentage.php" alt="Rolling Win Percentage Graph" class="rightimage" />
+<br />
+
+<h2>Balls</h2>
+<table width="100%" class="stats">
+<tr>
+<th>Color</th>
+<th>Ball Type</th>
+<th>Use Percentage</th>
+<th>Win Percentage</th>
+<th>Deuces Per Game</th>
+</tr>
+
+{section name=mysec loop=$balls}
+{strip}
+<tr>
+<th>{$balls[mysec].color}</th>
+<td>{$balls[mysec].type}</td>
+<td>{$balls[mysec].use_perc}</td>
+<td>{$balls[mysec].win_perc}</td>
+<td>{$balls[mysec].deuces_per_game}</td>
+</tr>
+{/strip}
+{/section}
+</table>
+
+{include file="footer.tpl"}
Index: templates/footer.tpl
===================================================================
--- templates/footer.tpl	(revision 0)
+++ templates/footer.tpl	(revision 0)
@@ -0,0 +1,8 @@
+<!-- content ends -->
+</div> <!--content-->
+<div id="copyright">
+&copy; 2004-2005
+</div>
+</div> <!--wrapper-->
+</body>
+</html>
Index: templates/flipcoin-results.tpl
===================================================================
--- templates/flipcoin-results.tpl	(revision 0)
+++ templates/flipcoin-results.tpl	(revision 0)
@@ -0,0 +1,21 @@
+{include file="header.tpl" title=$title}
+
+  <form action="" method="post">
+    <div id="labels">
+      <div class="widelabel" style="padding-bottom: 10px;">
+        <div>I want to go first:</div>
+        <input name="submit" type="submit" value="Go First" />
+      </div>
+      <div class="widelabel">
+{include file="choseballs.tpl"}
+      </div>
+      <div class="widelabel">
+        <div style="text-align: right; width: 25.7em; padding-top: 0px;">
+          <input name="submit" type="submit" value="Choose Balls" />
+        </div>
+      </div>
+    </div>
+  </form>
+
+
+{include file="footer.tpl"}
Index: templates/sidepanel.tpl
===================================================================
--- templates/sidepanel.tpl	(revision 0)
+++ templates/sidepanel.tpl	(revision 0)
@@ -0,0 +1,10 @@
+<div id="sidepanel">
+<div id="sp-buttons">
+<ul>
+<li><a href="startgame.php">Play a Game</a></li>
+<li><a href="addplayer.php">Add a Player</a></li>
+<li><a href="statistics.php">Get Statistics</a></li>
+<li class="last"><a href="allscores.php">Show All Scores</a></li>
+</ul>
+</div>
+</div>
Index: templates/addplayer.tpl
===================================================================
--- templates/addplayer.tpl	(revision 0)
+++ templates/addplayer.tpl	(revision 0)
@@ -0,0 +1,40 @@
+{include file="header.tpl" title=$title}
+
+{if $notice}
+<div id="notice">{$notice}</div>
+{/if}
+
+<form action="" method="post">
+  <div id="labels">
+    <div class="label">
+      <div>First Name:</div>
+      <input type="text" maxlength="25" size="25" name="fname" value="{$fname}" />
+    </div>
+    <div class="label">
+      <div>Last Name:</div>
+      <input type="text" maxlength="25" size="25" name="lname" value="{$lname}" />
+    </div>
+    <div class="label">
+      <div>User Name:</div>
+      <input type="text" maxlength="16" size="25" name="uname" value="{$uname}" />
+    </div>
+    <div class="label">
+      <div>Email Address:</div>
+      <input type="text" size="25" name="email" value="{$email}" />
+    </div>
+    <br />
+    <div class="label">
+      <div>Password:</div>
+      <input type="password" maxlength="25" size="25" name="pass" />
+    </div>
+    <div class="label">
+      <div>Verify Password:</div>
+      <input type="password" maxlength="25" size="25" name="vpass" />
+    </div>
+    <div class="submit">
+      <input class="submit" name="submit" type="submit" value="Submit" />
+    </div>
+  </div>
+</form>
+
+{include file="footer.tpl"}
Index: templates/index.tpl
===================================================================
--- templates/index.tpl	(revision 0)
+++ templates/index.tpl	(revision 0)
@@ -0,0 +1,8 @@
+{include file="header.tpl" title=$title}
+
+<div style="text-align: center">
+<p>This is meant to be an exercise in web development with php and sqlite.  This is also a great excuse to go extrememly overboard with statistics for bocce, but it should be trivial adding support for other database engines.</p>
+<img src="images/pope.gif" alt="The pope playing bocce" />
+</div>
+
+{include file="footer.tpl"}
Index: templates/flipping.tpl
===================================================================
--- templates/flipping.tpl	(revision 0)
+++ templates/flipping.tpl	(revision 0)
@@ -0,0 +1,12 @@
+{include file="header.tpl" title=$title}
+
+{literal}
+<script>
+function gotoFlipCoin() {
+  window.location="flipcoin.php";
+}
+setTimeout("gotoFlipCoin()", 2500);
+</script>
+{/literal}
+
+{include file="footer.tpl}
Index: templates/flipcoin-ballcolor.tpl
===================================================================
--- templates/flipcoin-ballcolor.tpl	(revision 0)
+++ templates/flipcoin-ballcolor.tpl	(revision 0)
@@ -0,0 +1,21 @@
+{include file="header.tpl" title=$title}
+
+<form action="" method="post">
+<div style="padding-top: 1em; padding-left: 1em;">{$chose_balls_name} chose to use the following balls:</div>
+<div style="padding-left: 6em;">
+{section name=ball loop=$prev_balls}
+{$prev_balls[ball].name}: {$prev_balls[ball].color}<br />
+{/section}
+</div>
+<div id="labels">
+<div class="label">
+{include file="choseballs.tpl"}
+</div>
+<div style="text-align: right; width: 24em; padding-top: 0px;">
+<input name="submit" type="submit" value="Choose Balls" />
+</div>
+</div>
+</form>
+
+
+{include file="footer.tpl"}
Index: templates/addscore-final.tpl
===================================================================
--- templates/addscore-final.tpl	(revision 0)
+++ templates/addscore-final.tpl	(revision 0)
@@ -0,0 +1,49 @@
+{include file="header.tpl" title=$title}
+
+<form action="" method="post">
+<input type="hidden" name="stage" value="gameon" />
+<center>
+<table class="score">
+<tr>
+<td><h1>{$winner_score}</h1></td>
+<td colspan="2"><h1>-</h1></td>
+<td><h1>{$loser_score}</h1></td>
+</tr>
+<tr>
+<td><h2>{$winner_name}</h2></td>
+<td colspan="2"></td>
+<td><h2>{$loser_name}</h2></td>
+</tr>
+</table>
+<br />
+<h3>Score History</h3>
+<table class="scoreboard">
+<tr>
+<td>{$winner_name}</td>
+{section name=hist loop=$win_hist}
+<td>{$win_hist[hist]}</td>
+{/section}
+</tr>
+<tr>
+<td>{$loser_name}</td>
+{section name=hist loop=$los_hist}
+<td>{$los_hist[hist]}</td>
+{/section}
+</tr>
+</table>
+
+<h3>Bruises</h3>
+<table class="bruises">
+<tr>
+<th>Name</th><th>Round</th><th>Success</th>
+</tr>
+{section name=br loop=$bruises}
+<tr>
+<td>{$bruises[br].name}</td><td>{$bruises[br].round}</td><td>{$bruises[br].suc}</td>
+</tr>
+{/section}
+</table>
+</center>
+</form>
+
+{include file="footer.tpl"}
Index: templates/onegame.tpl
===================================================================
--- templates/onegame.tpl	(revision 0)
+++ templates/onegame.tpl	(revision 0)
@@ -0,0 +1,44 @@
+{include file="header.tpl" title=$title}
+
+<h2>{$gamedate}</h2>
+<br />
+
+<h3>Score History</h3>
+<table class="scoreboard">
+<tr><th>Round</th>
+{section name=mysec loop=$scoredata}
+{strip}
+<th>{$scoredata[mysec].round}</th>
+{/strip}
+{/section}
+</tr>
+
+<tr><th>{$winner_name}</th>
+{section name=mysec loop=$scoredata}
+{strip}
+<td>{$scoredata[mysec].winner_pts}</td>
+{/strip}
+{/section}
+</tr>
+
+<tr><th>{$loser_name}</th>
+{section name=mysec loop=$scoredata}
+{strip}
+<td>{$scoredata[mysec].loser_pts}</td>
+{/strip}
+{/section}
+</tr>
+</table>
+
+<h3>Bruises</h3>
+<table class="bruises">
+<tr><th>Name</th><th>Round</th><th>Success</th></tr>
+{section name=mysec loop=$bruisedata}
+{strip}
+<tr>
+<td>{$bruisedata[mysec].name}</td><td>{$bruisedata[mysec].round}</td><td>{$bruisedata[mysec].success}</td>
+</tr>
+{/strip}
+{/section}
+</table>
+{include file="footer.tpl"}
Index: templates/flipcoin-gofirst.tpl
===================================================================
--- templates/flipcoin-gofirst.tpl	(revision 0)
+++ templates/flipcoin-gofirst.tpl	(revision 0)
@@ -0,0 +1,18 @@
+{include file="header.tpl" title=$title}
+
+  <form action="" method="post">
+    <div style="padding-top: 1em; width: 12em; text-align: right;">
+      {$chose_first_name} chose to go first.
+    </div>
+    <div id="labels">
+    <div class="widelabel">
+      {include file="choseballs.tpl"}
+    </div>
+    <div style="text-align: right; width: 25.7em; padding-top: 0px;">
+    <input name="submit" type="submit" value="Choose Balls" />
+    </div>
+    </div>
+  </form>
+
+
+{include file="footer.tpl"}
Index: templates/choseballs.tpl
===================================================================
--- templates/choseballs.tpl	(revision 0)
+++ templates/choseballs.tpl	(revision 0)
@@ -0,0 +1,10 @@
+<div id="balls-message">{$balls_message}</div>
+
+{section name=balltype loop=$balls}
+	<select name="ball{$balls[balltype].type}">
+		<option value="-----">{$balls[balltype].type_name}</option>
+		{section name=ball loop=$balls[balltype].balls}
+			<option value="{$balls[balltype].balls[ball].id}">{$balls[balltype].balls[ball].name}</option>
+		{/section}
+	</select>
+{/section}
Index: templates/test.tpl
===================================================================
--- templates/test.tpl	(revision 0)
+++ templates/test.tpl	(revision 0)
@@ -0,0 +1 @@
+blah{$name}ded
Index: addscore.php
===================================================================
--- addscore.php	(revision 17)
+++ addscore.php	(working copy)
@@ -19,379 +19,261 @@
  *Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
-session_start();
-require('db.php');
-require('boccelib.php');
+require('start.php');
 
 /* CurrentScore returns the total score of a player at any given time.
  * The score is only stored as in each round, not as a total score.  This
  * is done to avoid storing the same data in two places.  This function
  * just adds up all of the scores in each round and returns the total. */
 function CurrentScore($playernum) {
-  if (!isset($_SESSION["history$playernum"])) {
-    return 0;
-  }
+	if (!isset($_SESSION["history$playernum"])) {
+		return 0;
+	}
 
-  $score = 0;
-  foreach($_SESSION["history$playernum"] as $point) {
-    $score += $point;
-  }
+	$score = 0;
+	foreach($_SESSION["history$playernum"] as $point) {
+		$score += $point;
+	}
 
-  return $score;
+	return $score;
 }
 
 if (!isset($_SESSION["player1"]) or !isset($_SESSION["player2"])) {
-  redirect("startgame.php");
-  return 0;
+	redirect("startgame.php");
+	return 0;
 }
 
 if (!isset($_SESSION['history1'])) {
-  $_SESSION['history1'] = array();
-  $_SESSION['history2'] = array();
-  $_SESSION['bruises'] = array();
-  $_SESSION['round'] = 0;
+	$_SESSION['history1'] = array();
+	$_SESSION['history2'] = array();
+	$_SESSION['bruises'] = array();
+	$_SESSION['round'] = 0;
 }
 
 $confirmed = 0;
 
 if (isset($_POST["action"])) {
-  $round = $_SESSION['round'];
-  if ($_POST["action"] == "Player1Score1") {
-    $_SESSION["history1"][$round] = 1;
-    $_SESSION["history2"][$round] = 0;
-    $_SESSION['round']++;
-    $_SESSION['pallino_toss'][$round + 1] = $_SESSION['player1'];
-  } elseif ($_POST["action"] == "Player1Score2") {
-    $_SESSION["history1"][$round] = 2;
-    $_SESSION["history2"][$round] = 0;
-    $_SESSION['round']++;
-    $_SESSION['pallino_toss'][$round + 1] = $_SESSION['player1'];
-  } elseif ($_POST["action"] == "Player2Score1") {
-    $_SESSION["history2"][$round] = 1;
-    $_SESSION["history1"][$round] = 0;
-    $_SESSION['round']++;
-    $_SESSION['pallino_toss'][$round + 1] = $_SESSION['player2'];
-  } elseif ($_POST["action"] == "Player2Score2") {
-    $_SESSION["history2"][$round] = 2;
-    $_SESSION["history1"][$round] = 0;
-    $_SESSION['round']++;
-    $_SESSION['pallino_toss'][$round + 1] = $_SESSION['player2'];
-  } elseif ($_POST["action"] == "Player1GoodBruise") {
-    $_SESSION['bruises'][] = array($_SESSION['player1'],$round,1);
-  } elseif ($_POST["action"] == "Player1BadBruise") {
-    $_SESSION['bruises'][] = array($_SESSION['player1'],$round,0);
-  }  elseif ($_POST["action"] == "Player2GoodBruise") {
-    $_SESSION['bruises'][] = array($_SESSION['player2'],$round,1);
-  } elseif ($_POST["action"] == "Player2BadBruise") {
-    $_SESSION['bruises'][] = array($_SESSION['player2'],$round,0);
-  } elseif ($_POST["action"] == "confirm") {
-    $confirmed = 1;
-  } elseif ($_POST["action"] == "undo") {
-    //undo bruise or last scored point
-    $bruiseInThisRound = false;
-    foreach($_SESSION['bruises'] as $bruise) {
-      if ($bruise[1] == $round) {
-	$bruiseInThisRound = true;
-	break;
-      }
-    }
+	$round = $_SESSION['round'];
+	if ($_POST["action"] == "Player1Score1") {
+		$_SESSION["history1"][$round] = 1;
+		$_SESSION["history2"][$round] = 0;
+		$_SESSION['round']++;
+		$_SESSION['pallino_toss'][$round + 1] = $_SESSION['player1'];
+	} elseif ($_POST["action"] == "Player1Score2") {
+		$_SESSION["history1"][$round] = 2;
+		$_SESSION["history2"][$round] = 0;
+		$_SESSION['round']++;
+		$_SESSION['pallino_toss'][$round + 1] = $_SESSION['player1'];
+	} elseif ($_POST["action"] == "Player2Score1") {
+		$_SESSION["history2"][$round] = 1;
+		$_SESSION["history1"][$round] = 0;
+		$_SESSION['round']++;
+		$_SESSION['pallino_toss'][$round + 1] = $_SESSION['player2'];
+	} elseif ($_POST["action"] == "Player2Score2") {
+		$_SESSION["history2"][$round] = 2;
+		$_SESSION["history1"][$round] = 0;
+		$_SESSION['round']++;
+		$_SESSION['pallino_toss'][$round + 1] = $_SESSION['player2'];
+	} elseif ($_POST["action"] == "Player1GoodBruise") {
+		$_SESSION['bruises'][] = array($_SESSION['player1'],$round,1);
+	} elseif ($_POST["action"] == "Player1BadBruise") {
+		$_SESSION['bruises'][] = array($_SESSION['player1'],$round,0);
+	}  elseif ($_POST["action"] == "Player2GoodBruise") {
+		$_SESSION['bruises'][] = array($_SESSION['player2'],$round,1);
+	} elseif ($_POST["action"] == "Player2BadBruise") {
+		$_SESSION['bruises'][] = array($_SESSION['player2'],$round,0);
+	} elseif ($_POST["action"] == "confirm") {
+		$confirmed = 1;
+	} elseif ($_POST["action"] == "undo") {
+		//undo bruise or last scored point
+		$bruiseInThisRound = false;
+		foreach($_SESSION['bruises'] as $bruise) {
+			if ($bruise[1] == $round) {
+				$bruiseInThisRound = true;
+				break;
+			}
+		}
 
-    if ($bruiseInThisRound) {
-      $_SESSION["bruises"] = array_slice($_SESSION["bruises"], 0,
-					 count($_SESSION["bruises"]) - 1);
-    } else {
-      $_SESSION["history1"] = array_slice($_SESSION["history1"], 0,
-					  count($_SESSION["history1"]) - 1);
-      $_SESSION["history2"] = array_slice($_SESSION["history2"], 0,
-					  count($_SESSION["history2"]) - 1);
-      $_SESSION['round']--;
-    }
-  }
- }
+		if ($bruiseInThisRound) {
+			$_SESSION["bruises"] = array_slice($_SESSION["bruises"], 0,
+					count($_SESSION["bruises"]) - 1);
+		} else {
+			$_SESSION["history1"] = array_slice($_SESSION["history1"], 0,
+					count($_SESSION["history1"]) - 1);
+			$_SESSION["history2"] = array_slice($_SESSION["history2"], 0,
+					count($_SESSION["history2"]) - 1);
+			$_SESSION['round']--;
+		}
+	}
+}
 
 $score1 = CurrentScore(1);
 $score2 = CurrentScore(2);
 
 if (!$confirmed) {
-  
-  //displaying confirmation page if the game would end
-  $confirmPage = false;
-  if ($score1 >= 7) {
-    $score1 = 7;
-    $confirmPage = true;
-  }
-  if ($score2 >= 7) {
-    $score2 = 7;
-    $confirmPage = true;
-  }
+	//displaying confirmation page if the game would end
+	$confirmPage = false;
+	if ($score1 >= 7) {
+		$score1 = 7;
+		$confirmPage = true;
+	}
+	if ($score2 >= 7) {
+		$score2 = 7;
+		$confirmPage = true;
+	}
 
-  $names = getNames($_SESSION['player1'], $_SESSION['player2']);
-  $name1 = $names[0];
-  $name2 = $names[1];
+	$names = getNames($_SESSION['player1'], $_SESSION['player2']);
+	$name1 = $names[0];
+	$name2 = $names[1];
 
-  require("header.php");
-  require("side.php");
-?>
-<div class="body">
-<form action="" method="post">
-<center>
-<table>
-<tr>
-<td><h1 class="score"><?=$score1?></h1></td>
-<td><h1 class="score">-</h1></td>
-<td><h1 class="score"><?=$score2?></h1></td>
-</tr>
-<tr>
-<td><h2><?=$name1?></h2></td>
-<td></td>
-<td><h2><?=$name2?></h2></td>
-</tr>
-</table>
-<br />
-<table class="controls">
-<?
-   if (!$confirmPage) {
-?>
-<tr><td>
-<table class="controlpad">
-<tr>
-<td><?=$name1?></td>
-<td><button type="submit" name="action" value="Player1Score1">Score 1</button></td>
-<td><button type="submit" name="action" value="Player1Score2" >Deuce</button></td>
-<td><button type="submit" name="action" value="Player1GoodBruise" >Successful Bruise</button></td>
-<td><button type="submit" name="action" value="Player1BadBruise" >Missed Bruise</button></td>
-</tr>
-<tr>
-<td><?=$name2?></td>
-<td><button type="submit" name="action" value="Player2Score1">Score 1</button></td>
-<td><button type="submit" name="action" value="Player2Score2" >Deuce</button></td>
-<td><button type="submit" name="action" value="Player2GoodBruise" >Successful Bruise</button></td>
-<td><button type="submit" name="action" value="Player2BadBruise" >Missed Bruise</button></td>
-</tr>
-</table>
-</td>
-</tr>
-<?
-   }
+	$page->assign('score1', $score1);
+	$page->assign('score2', $score2);
+	$page->assign('name1', $name1);
+	$page->assign('name2', $name2);
 
-   if ((count($_SESSION['history1']) > 0) or (count($_SESSION['bruises']))) {
-     print '<tr><td colspan="5"><button type="submit" name="action" value="undo">Undo</button></td></tr>';
-   }
-   if ($confirmPage) {
-     print '<tr><td colspan="5"><button type="submit" name="action" value="confirm">Confirm</button></td></tr>';
-   }
+	$page->assign('confirmPage', $confirmPage);
 
-?>
-</table>
+	if ((count($_SESSION['history1']) > 0) or (count($_SESSION['bruises']))) {
+		$page->assign('undo', 1);
+	}
 
-<?
-   if (count($_SESSION['history1'])) {
-     print '<h3>Score History</h3>';
-     print '<table class="scoreboard">';
-     print '<tr>';
+	if (count($_SESSION['history1'])) {
+		$hist1 = Array();
+		foreach ($_SESSION['history1'] as $history) {
+			$hist1[] = $history;
+		}
+		$page->assign('hist1', $hist1);
      
-     print "<td>$name1</td>";
-     foreach ($_SESSION['history1'] as $history) {
-       print "<td>$history</td>";
-     }
-     
-     print '</tr>';
-     print '<tr>';
-     
-     print "<td>$name2</td>";
-     foreach ($_SESSION['history2'] as $history) {
-       print "<td>$history</td>";
-     }
-     print '</tr>';
-     print '</table>';
-   }
+		$hist2 = Array();
+		foreach ($_SESSION['history2'] as $history) {
+			$hist2[] = $history;
+		}
+		$page->assign('hist2', $hist2);
+		$page->assign('showhistory', 1);
+	}
 
-  if (count($_SESSION['bruises'])) {
-    print '<h3>Bruises</h3>';
-    print '<table class="bruises">';
-
-    print '<th>Name</th><th>Round</th><th>Success</th>';
-
-    foreach($_SESSION['bruises'] as $bruise) {
-      print '<tr>';
-      if ($bruise[0] == $_SESSION['player1']) {
-	$name = $name1;
-      } else {
-	$name = $name2;
-      }
-      print "<td>$name</td>";
-      $round = $bruise[1] + 1;
-      print "<td>" . $round . "</td>";
-      if ($bruise[2] == 1) {
-	$success = "Successful";
-      } else {
-	$success = "Failure";
-      }
-      print "<td>$success</td>";
-      print "</tr>";
-    }
-
-    print '</table>';
-  }
-?>
-
-</center>
-</form>
-</div>
-</body>
-</html>
-<?
-  } else {
+	if (count($_SESSION['bruises'])) {
+		$bruises = Array();
+		foreach($_SESSION['bruises'] as $bruise) {
+			if ($bruise[0] == $_SESSION['player1']) {
+				$name = $name1;
+			} else {
+				$name = $name2;
+			}
+			$bruises[] = Array(
+				'name' => $name,
+				'round' => $bruise[1] + 1,
+				'succ' => (($bruise[2] == 1)?"Successful":"Failure"));
+		}
+	}
+	$page->assign('bruises', $bruises);
+	$page->assign('showbruises', 1);
+	$template = "progress";
+} else {
+	//if either score is greater than 7 and confirmed == true
   
-    //if either score is greater than 7 and confirmed == true
-  
-    $names = getNames($_SESSION['player1'], $_SESSION['player2']);
-    $name1 = $names[0];
-    $name2 = $names[1];
+	$names = getNames($_SESSION['player1'], $_SESSION['player2']);
+	$name1 = $names[0];
+	$name2 = $names[1];
 
-    db_open();
-    $winner_score = 7;
-    $coinflip_winner = $_SESSION["coinflip_winner"];
+	db_open();
+	$winner_score = 7;
+	$coinflip_winner = $_SESSION["coinflip_winner"];
   
-    if ($score1 >= 7) {
-      $winner = $_SESSION['player1'];
-      $loser = $_SESSION['player2'];
-      $loser_score = $score2;
-      $winner_history = $_SESSION['history1'];
-      $loser_history = $_SESSION['history2'];
-      $winner_name = $name1;
-      $loser_name = $name2;
-      $winner_ball1=$_SESSION['player1_ball1'];
-      $winner_ball2=$_SESSION['player1_ball2'];
-      $loser_ball1=$_SESSION['player2_ball1'];
-      $loser_ball2=$_SESSION['player2_ball2'];
-    } else {
-      $winner = $_SESSION['player2'];
-      $loser = $_SESSION['player1'];
-      $loser_score = $score1;
-      $winner_history = $_SESSION['history2'];
-      $loser_history = $_SESSION['history1'];
-      $winner_name = $name2;
-      $loser_name = $name1;
-      $winner_ball1=$_SESSION['player2_ball1'];
-      $winner_ball2=$_SESSION['player2_ball2'];
-      $loser_ball1=$_SESSION['player1_ball1'];
-      $loser_ball2=$_SESSION['player1_ball2'];
-    }
+	if ($score1 >= 7) {
+		$winner = $_SESSION['player1'];
+		$loser = $_SESSION['player2'];
+		$loser_score = $score2;
+		$winner_history = $_SESSION['history1'];
+		$loser_history = $_SESSION['history2'];
+		$winner_name = $name1;
+		$loser_name = $name2;
+		$winner_ball1=$_SESSION['player1_ball1'];
+		$winner_ball2=$_SESSION['player1_ball2'];
+		$loser_ball1=$_SESSION['player2_ball1'];
+		$loser_ball2=$_SESSION['player2_ball2'];
+	} else {
+		$winner = $_SESSION['player2'];
+		$loser = $_SESSION['player1'];
+		$loser_score = $score1;
+		$winner_history = $_SESSION['history2'];
+		$loser_history = $_SESSION['history1'];
+		$winner_name = $name2;
+		$loser_name = $name1;
+		$winner_ball1=$_SESSION['player2_ball1'];
+		$winner_ball2=$_SESSION['player2_ball2'];
+		$loser_ball1=$_SESSION['player1_ball1'];
+		$loser_ball2=$_SESSION['player1_ball2'];
+	}
 
-#    $query = "insert into game values(null, $loser,$winner,$loser_score,$winner_score,datetime('now'))";
-    $query = "insert into game values(null, $loser,$winner,$coinflip_winner,$loser_score,$winner_score,'$winner_ball1','$winner_ball2','$loser_ball1','$loser_ball2', datetime('now'))";
-    $result = db_query($query);
-    $game_id = db_last_insert_rowid();
+	$query = "insert into game values(null, $loser,$winner,$coinflip_winner,$loser_score,$winner_score,'$winner_ball1','$winner_ball2','$loser_ball1','$loser_ball2', datetime('now'))";
+	$result = db_query($query);
+	$game_id = db_last_insert_rowid();
     
-    //maybe cut this down to one long transaction
-    for ($round = 0; $round < count($winner_history); $round++) {
-      if ($winner_history[$round] != 0) {
-	$amount = $winner_history[$round];
-	$scorer = $winner;
-      } else {
-	$amount = $loser_history[$round];
-	$scorer = $loser;
-      }
-      $pallino_tosser = $_SESSION['pallino_toss'][$round];
-      $roundnumber = $round + 1;
-      $query = "insert into point values(null, $game_id, $roundnumber, $scorer, $pallino_tosser, $amount)";
-      $result = db_query($query);
-    }
+	//maybe cut this down to one long transaction
+	for ($round = 0; $round < count($winner_history); $round++) {
+		if ($winner_history[$round] != 0) {
+			$amount = $winner_history[$round];
+			$scorer = $winner;
+		} else {
+			$amount = $loser_history[$round];
+			$scorer = $loser;
+		}
+		$pallino_tosser = $_SESSION['pallino_toss'][$round];
+		$roundnumber = $round + 1;
+		$query = "insert into point values(null, $game_id, $roundnumber, $scorer, $pallino_tosser, $amount)";
+		$result = db_query($query);
+	}
     
-    //store bruises in database
-    foreach($_SESSION['bruises'] as $bruise) {
-      $player = $bruise[0];
-      $roundnum = $bruise[1];
-      $success = $bruise[2];
+	//store bruises in database
+	foreach($_SESSION['bruises'] as $bruise) {
+		$player = $bruise[0];
+		$roundnum = $bruise[1];
+		$success = $bruise[2];
 
-      $query = "insert into bruise values(null, $game_id, $roundnum, $player, $success)";
-      $result = db_query($query);
-    }
+		$query = "insert into bruise values(null, $game_id, $roundnum, $player, $success)";
+		$result = db_query($query);
+	}
 
-    db_close();
+	db_close();
     
-require("header.php");
-require("side.php");
-?>
-<div class="body">
-<form action="" method="post">
-<input type="hidden" name="stage" value="gameon" />
-<center>
-<h1 class="title">Final Score</h1><br />
-<table>
-<tr>
-<td><h1><?=$winner_score?></h1></td>
-<td colspan="2"><h1>-</h1></td>
-<td><h1><?=$loser_score?></h1></td>
-</tr>
-<tr>
-<td><h2><?=$winner_name?></h2></td>
-<td colspan="2"></td>
-<td><h2><?=$loser_name?></h2></td>
-</tr>
-</table>
-<br />
-<h3>Score History</h3>
-<table class="scoreboard">
-<tr>
-<?
-print "<td>$winner_name</td>";
-foreach ($winner_history as $history) {
-    print "<td>$history</td>";
-}
+	$page->assign('subtitle', "Final Score");
 
-?>
-</tr>
-<tr>
-<?
-print "<td>$loser_name</td>";
-foreach ($loser_history as $history) {
-    print "<td>$history</td>";
-}
+	$page->assign('winner_score', $winner_score);
+	$page->assign('loser_score', $loser_score);
+	$page->assign('winner_name', $winner_name);
+	$page->assign('loser_name', $loser_name);
 
-?>
-</tr>
-</table>
-<?
-  if (count($_SESSION['bruises'])) {
-    print '<h3>Bruises</h3>';
-    print '<table class="bruises">';
+	$win_hist = Array();
+	foreach ($winner_history as $history) {
+		$win_hist[] = $history;
+	}
+	$page->assign('win_hist', $win_hist);
 
-    print '<th>Name</th><th>Round</th><th>Success</th>';
+	$los_hist = Array();
+	foreach ($loser_history as $history) {
+		$los_hist[] = $history;
+	}
+	$page->assign('los_hist', $los_hist);
 
-    foreach($_SESSION['bruises'] as $bruise) {
-      print '<tr>';
-      if ($bruise[0] == $_SESSION['player1']) {
-	$name = $name1;
-      } else {
-	$name = $name2;
-      }
-      print "<td>$name</td>";
-      $round = $bruise[1] + 1;
-      print "<td>" . $round . "</td>";
-      if ($bruise[2] == 1) {
-	$success = "Successful";
-      } else {
-	$success = "Failure";
-      }
-      print "<td>$success</td>";
-      print "</tr>";
-    }
+	if (count($_SESSION['bruises'])) {
+		$bruises = Array();
+		foreach($_SESSION['bruises'] as $bruise) {
+			if ($bruise[0] == $_SESSION['player1']) {
+				$name = $name1;
+			} else {
+				$name = $name2;
+			}
+			$bruises[] = Array(
+			      	'name' => $name,
+				'round' => $bruise[1] + 1,
+				'suc' => (($bruise[2] == 1)?"Successful":"Failure"));
+		}
+	}
+	$page->assign('bruises', $bruises);
+	$template = "final";
+}
 
-    print '</table>';
-  }
-?>
-</center>
-</form>
-</div>
-</body>
-</html>
+$page->display("addscore-$template.tpl");
 
-<?
-/*
-    }
-*/
-
-}
 ?>
Index: main.css
===================================================================
--- main.css	(revision 17)
+++ main.css	(working copy)
@@ -1,174 +1,187 @@
+/* very global settings */
 body {
-  font-family: Arial;
+	font-family: Arial;
 }
 
-div#header { 
-  background-color: #6495ED;
-  border: 1px solid #000;
-  margin: 0;
+/* just the main output area */
+#wrapper {
 }
 
-span.title { 
-  font-size: xx-large;
-  font-weight: bold;
+/* the header */
+#header {
+	background-color: #6495ED;
+	border: 1px solid #000000;
+	margin: 0px;
 }
 
-span#maintitle {
-  color: #191970;
-  position: relative;
-  left: 50px;
-  bottom: 50px;
+/* the title, part of the header */
+#title {
+	color: #191970;
+	font-size: xx-large;
+	font-weight: bold;
+	position: relative;
+	left: 50px;
+	bottom: 50px;
 }
 
-div#sidepanel { 
-  position: relative;
-  float: left;
-  top: 10px;
-  margin:0;
-  background-color: #6495ED;
-  border: 1px solid #000;
-  width: 180px;
+/* sidepanel */
+#sidepanel {
+	width: 180px;
+	border: 1px solid #000000;
+	background-color: #6495ED;
+	margin: 0px;
+	float: left;
+	position: relative;
+	top: 10px;
 }
 
-div#sidepanel div#buttons {
-  margin: 0px;
-  width: 180px;
-  background-color: #6495ED;
+/* sidepanel buttons */
+/*#sp-buttons {
+	margin: 0px;
+	width: 180px;
+	background-color: #6495ED;
+}*/
+
+/* sidepanel list style */
+#sp-buttons ul {
+	list-style: none;
+	margin: 0px;
+	padding: 0px;
+	border: none;
 }
 
-div#sidepanel div#buttons ul {
-  list-style: none;
-  margin: 0;
-  padding: 0;
-  border: none;
+/* sidepanel list item */
+#sp-buttons li {
+	border-bottom: 1px solid #90BADE;
+	margin: 0px;
 }
 
-div#sidepanel div#buttons li {
-  border-bottom: 1px solid #90bade;
-  margin: 0;
+/* sidepanel last list item */
+#sp-buttons li.last {
+	border-bottom: none;
 }
 
-div#sidepanel div#buttons li.last {
-  border-bottom: 0;
+/* nice sidepanel links */
+#sp-buttons a {
+	display: block;
+	padding: 5px 15px 5px 15px;
+	color: #191970;
+	text-decoration: none;
 }
 
-div#sidepanel div#buttons li a {
-  display: block;
-  padding: 5px 15px 5px 15px;
-  background-color: #6495ED;
-  color: #191970;
-  text-decoration: none;
+/* javascript is dead! */
+#sp-buttons a:hover {
+	background-color: #00008B;
+	color: #ffffff;
 }
 
-div#sidepanel div#buttons li a:hover {
-  background-color: #00008B;
-  color: #fff;
+/* the content pane */
+#content {
+	margin-left: 200px;
+	margin-top: 10px;
+	width: auto;
 }
 
-html>body #button li a {
-  width: auto;
+/* paragraph style */
+#content p {
+	margin: 1em 3em 1em 3em;
 }
 
-div#mainbody { 
-  text-align: center;
+/* subtitle */
+#subtitle {
 }
 
-div#mainbody p { 
-  margin: 1em 3em 1em 3em;
+/* subtitle - the text */
+#subtitle h1 {
+	margin: 0px;
+	//margin-bottom: 10pt;
 }
 
-div.body { 
-  margin-left: 200px;
-  margin-top: 10px;
+#subtitlehint {
+	padding-left: 10px;
+	font-size: small;
+	font-weight: normal;
+	margin-bottom: 1em;
 }
 
-img { 
-  border:0;
+h2, h3 {
+	margin: 0px;
 }
 
-td {
-text-align: center;
+table.score,
+table.controls {
+	border: 0px;
 }
 
-h1, h2, h3 {
-padding: 0; margin: 0;
+table.score td {
+	border: 0px;
 }
 
-h1 span.titlehint {
-padding-left: 10px;
-font-size: small;
-font-weight: normal;
+#labels {
+	margin-top: 5px;
 }
 
-h1.score {
-font-size: 48pt;
+div.label,
+div.widelabel {
+	padding-top: 3px;
 }
 
-div.label {
-padding-top: 3px;
+div.submit {
+	padding-left: 10.5em;
 }
 
-div.label div {
-text-align: right;
-float: left;
-width: 10em;
-padding-right: 5px;
+#notice {
+	margin-top: 10px;
+	color: red;
+	text-align: center;
+	width: 100%;
 }
 
-div.label#password {
-margin-top: 20px;
+div.label div,
+div.widelabel div {
+	text-align: right;
+	float: left;
+	width: 10em;
+	padding-right: 5px;
 }
 
-div.labels div.submit {
-margin-top: 5px;
-margin-left: 10.5em;
+div.widelabel div {
+	width: 12em;
 }
 
-div.labels {
-margin-top: 10px;
+/* no border around images */
+img {
+	border: none;
 }
 
-div.notice {
-margin-top: 10px;
-color: red;
-text-align: center;
-width: 100%;
+img.leftimage {
+	float: left;
 }
 
-table.controls {
-padding: 10px;
+img.rightimage {
+	float:right;
 }
 
-table.stats,
-table.controlpad,
-table.scoreboard,
-table.bruises,
-table.allgames {
-border: thin solid black;
-border-collapse: collapse;
-margin-bottom: 10px;
+table {
+	border: thin solid black;
+	border-collapse: collapse;
+	margin-bottom: 10px;
 }
 
-table.stats td,
-table.stats th,
-table.controlpad td,
-table.scoreboard td,
-table.scoreboard th,
-table.bruises td,
-table.bruises th,
-table.allgames td,
-table.allgames th {
-border: thin solid black;
-padding: 5px;
+table.stats {
+	font-size: small;
 }
 
-table.stats td,
-table.stats th {
-font-size: small;
+th, td {
+	border: thin solid black;
+	text-align: center;
+	padding: 5px;
 }
 
-table#graphs img {
-padding-right: 10px;
-padding-bottom: 8px;
+#copyright {
+	border: none;
+	text-align: center;
+	font-size: 10pt;
+	color: #AAAAAA;
+	margin-top: 1em;
 }
-
Index: startgame.php
===================================================================
--- startgame.php	(revision 17)
+++ startgame.php	(working copy)
@@ -19,9 +19,7 @@
  *Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
  */
 
-session_start();
-require('db.php');
-require('boccelib.php');
+require('start.php');
 
 $uname = "";
 $titlehint = "";
@@ -110,41 +108,11 @@
   $playernum = "Second";
 }
 
-require("header.php");
-require("side.php");
+$page->assign('notice', $notice);
+$page->assign('subtitle', "Login $playernum Player");
+$page->assign('subtitlehint', $titlehint);
+$page->assign('uname', $uname);
 
-print "<div class=\"body\">";
+$page->display('startgame.tpl');
 
-if ($notice != "") {
-  print "<div class=\"notice\">$notice</div>";
-}
-
 ?>
-<h1 class="title">Login <?=$playernum?> Player<span class="titlehint"><?=$titlehint?></span></h1>
-<form action="" method="post">
-  <div class="labels">
-    <div class="label">
-      <div>User Name:</div>
-      <input type="text" maxlength="16" size="25" name="uname" value="<?=$uname?>" />
-    </div>
-    <div class="label">
-      <div>Password:</div>
-      <input type="password" maxlength="25" size="25" name="pass" />
-    </div>
-    <div class="submit">
-      <input name="submit" type="submit" value="Login" />
-      <input name="submit" type="submit" value="Cancel" />
-    </div>
-  </div>
-</form>
-</div>
-
-<script type="text/javascript">
-function setfocus() {
-  document.forms[0].uname.focus();
-}
-window.onload=setfocus;
-</script>
-
-</body>
-</html>
